### ISSUE TITLE
Compile-on-save with typescriptservices.js from LKG drops module prefixes

### ISSUE THREAD

**Issue by NoelAbrahams** [Microsoft/TypeScript#2061]
Hi,

I'm using VS 2015 CTP 5 and wanted to try out `typescriptservices.js` from LKG of 12-Feb-2015.

Code like the following:

```typescript
// my.d.ts
declare module foo.bar {
  class Baz { }
}
```

When referenced from another project 
```typescript
/// <reference path='my.d.ts' />
module foo.bar {
      var baz = <any>Baz.prototype;
}
```

is compiling into

```javascript
var foo;
(function (foo) {
    var bar;
    (function (bar) {
        var baz = /*bar.*/Baz.prototype; // missing prefix
    })(bar = foo.bar || (foo.bar = {}));
})(foo || (foo  = {}));
```

This is only a problem when compile is triggered via save - generates the correct JavaScript when building.

**Edit**

This is a little bit more involved: The `combine JS into output file` also needs to be set. The faulty JS is generated not when the code file itself is saved, but when a sibling file in the same project is saved (which would cause the combined file to be generated).

**Comment by NoelAbrahams** [comment]
This bug is present in 1.5 beta. I would have thought it's critical as it produces JavaScript that fails at runtime.

**Comment by mhegazy** [comment]
@JsonFreeman  can you take a look.

**Comment by JsonFreeman** [comment]
@NoelAbrahams do you have noEmitOnError explicitly set to false in your project? I've investigated this with @vladima, but the only way we were able to reproduce it is to set noEmitOnError to false.

**Comment by mhegazy** [comment]
I got it to repro earlier today without any changes. You just need to create a project add two files as listed above set --out, touch both files and then save the one with declare module.

**Comment by vladima** [comment]
@mhegazy  That is **very** interesting. What is special in scenario with '--out' is that you''ll trigger save in one file but during emit you'll examine the entire project. Crux of the problem: initial file will be typechecked however other files -- not so `nodeLinks`\`symbolLinks` for them will not be populated -> incorrect emit. Why it works today: see [line 215 in program.ts](https://github.com/Microsoft/TypeScript/blob/master/src/compiler/program.ts#L215) - here we call `getPreEmitDiagnostics` that will query semantic diagnostics for all files and so all files will be typechecked. However this function is called only if `noEmitOnError` flag is set. If it is not set then emit will almost always be incorrect and we've verified that.

**Comment by mhegazy** [comment]
That makes sense.. `--noEmitOnError` is set in projects by default...

**Comment by JsonFreeman** [comment]
@mhegazy This brings us back to the question, how did it repro for you if you had noEmitOnError turned on?

**Comment by JsonFreeman** [comment]
@NoelAbrahams please try upgrading your VS with my fix, and see if it fixes your problem.
